server/src/test/java/org/elasticsearch/common/settings/SettingTests.java

```diff
@@ -1011,4 +1013,15 @@ public void testAffixMapUpdateWithNullSettingValue() {
         assertEquals("", value);
     }
 
+    @Test
+    public void test_empty_version_from_setting() {
+        var emptyVersion = IndexMetadata.SETTING_INDEX_VERSION_CREATED.get(Settings.EMPTY);
+        assertThat(emptyVersion, is((Version.V_EMPTY)));
+    }
+
+    @Test
```

Comment:
Would be good to add a comment here that we had a regression where the defaultValue was not handled correctly